daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -520,6 +508,10 @@ func (l *logStream) publishBatch(events []wrappedEvent) {
 	} else {
 		l.sequenceToken = nextSequenceToken
 	}
+	// Reset occurs regardless of error. The payload itself may be
+	// causing issues and would otherwise prevent more events from being
+	// sent.
+	batch.Reset()
```

Comment:
Should `batch.Reset()` happen here or in `processEvent` after the call to `publishBatch`?  My vote would probably be for the latter since `processEvent` is what owns the `eventBatch`.

Reply:
This would make it clearer that we're reusing the object if we lift the call up as you suggest - note that the batch is owned by `collectBatch` where additional call sites would exist. This change would mean that each `publishBatch()` _must_ be followed by `batch.Reset()`, otherwise duplicate batch of events will be put into CloudWatch. It's simpler to leave it in `publishBatch`, but this hides its side effect; I'm going to add the additional calls to `batch.Reset()` unless you think that doesn't make sense or if you had something else in mind.

Reply:
A different model (though I haven't thought about it enough to know whether I like the idea) is to make a new method `batch.Publish()` that is responsible for both publishing and resetting the internal buffer.

Reply:
I had a similar thought, but I dismissed on the grounds of keeping the submission of the events in the log driver and out of the `eventBatch` container type. I'd rather have the submission responsibility on the log driver for now.